app/scripts/directives/events.js

```diff
@@ -80,64 +78,90 @@ angular.module('openshiftConsole')
 
         // Invoke update when first called, debouncing subsequent calls.
         var debounceUpdate = _.debounce(function() {
-          $scope.$evalAsync(update);
+          if (!allEvents) {
+            return;
+          }
+
+          $scope.$evalAsync(function() {
+            // Assign `$scope.events` here instead of the watch callback to
+            // avoid the "Filter hiding all events" message from flickering
+            // briefly before the debounced filter and update calls.
+            $scope.events = filterForResource(allEvents);
+            update();
+          });
         }, 250, {
           leading: true,
-          trailing: false,
+          trailing: true,
```

Comment:
Not sure why trailing was false. I think it should be true.

https://lodash.com/docs/3.10.1#debounce

> Note: If leading and trailing options are true, func is invoked on the trailing edge of the timeout only if the the debounced function is invoked more than once during the wait timeout.